dockstore-webservice/src/main/java/io/dockstore/webservice/core/Entry.java

```diff
@@ -119,7 +123,11 @@
     @NamedQuery(name = "io.dockstore.webservice.core.Entry.findLabelByEntryId", query = "SELECT e.labels FROM Entry e WHERE e.id = :entryId"),
     @NamedQuery(name = "Entry.findToolsDescriptorTypes", query = "SELECT t.descriptorType FROM Tool t WHERE t.id = :entryId"),
     @NamedQuery(name = "Entry.findWorkflowsDescriptorTypes", query = "SELECT w.descriptorType FROM Workflow w WHERE w.id = :entryId"),
-    @NamedQuery(name = "Entry.findAllGitHubEntriesWithNoTopicAutomatic", query = "SELECT e FROM Entry e WHERE e.gitUrl LIKE 'git@github.com%' AND e.topicAutomatic IS NULL")
+    @NamedQuery(name = "Entry.findAllGitHubEntriesWithNoTopicAutomatic", query = "SELECT e FROM Entry e WHERE e.gitUrl LIKE 'git@github.com%' AND e.topicAutomatic IS NULL"),
+    @NamedQuery(name = ENTRY_GET_EXECUTION_METRIC_PARTNERS, query = "select new io.dockstore.webservice.core.database.PartnerMetrics(v.parent.id, KEY(v.metricsByPlatform)) from Version v "
+            + "where KEY(v.metricsByPlatform) != io.dockstore.common.Partner.ALL and value(v.metricsByPlatform).executionStatusCount != null and v.parent.id in (:entryIds) group by v.parent.id, key(v.metricsByPlatform)"),
+    @NamedQuery(name = ENTRY_GET_VALIDATION_METRIC_PARTNERS, query = "select new io.dockstore.webservice.core.database.PartnerMetrics(v.parent.id, KEY(v.metricsByPlatform)) from Version v "
+            + "where KEY(v.metricsByPlatform) != io.dockstore.common.Partner.ALL and value(v.metricsByPlatform).validationStatus != null and v.parent.id in (:entryIds) group by v.parent.id, key(v.metricsByPlatform)")
```

Comment:
I'm a little bit concerned how these are going to perform when there's tens of thousands of metrics data points per entry, but we can always optimize later, if necessary.

Reply:
Fair point, I wonder if we shouldn't have an index on version_metrics.platform. Not a SQL expert, but it might help with the group by.

Reply:
> I'm a little bit concerned how these are going to perform when there's tens of thousands of metrics data points per entry, but we can always optimize later, if necessary.

For clarification, there shouldn't be tens of thousands of metrics data points per entry in the database. The individual datapoints are stored in S3 (where it might reach the tens of thousands) and the metrics aggregator aggregates the data in S3 into aggregated metrics to put into the database. The number of entries in a Version's `metricsByPlatform` map would have max the number of enums in [Partner](https://github.com/dockstore/dockstore/blob/b16138467ebe0a09944bcae65353a0a8168ba950/dockstore-common/src/main/java/io/dockstore/common/Partner.java#L24-L36)

Reply:
> > I'm a little bit concerned how these are going to perform when there's tens of thousands of metrics data points per entry, but we can always optimize later, if necessary.
> 
> For clarification, there shouldn't be tens of thousands of metrics data points per entry in the database. The individual datapoints are stored in S3 (where it might reach the tens of thousands) and the metrics aggregator aggregates the data in S3 into aggregated metrics to put into the database. The number of entries in a Version's `metricsByPlatform` map would have max the number of enums in [Partner](https://github.com/dockstore/dockstore/blob/b16138467ebe0a09944bcae65353a0a8168ba950/dockstore-common/src/main/java/io/dockstore/common/Partner.java#L24-L36)

Makes sense, concern satisfied. :)